### PR TITLE
 chore(tests): added some more widget related tests

### PR DESCRIPTION
**Pull request by jdalsem**

### CODE REVIEW

File: engine/lib/widgets.php [hunk 1]
```diff
@@ -303,46 +305,50 @@ function _elgg_create_default_widgets($event, $type, $entity) {
...
-					'subtype' => 'widget',
-					'owner_guid' => elgg_get_site_entity()->guid,
-					'private_setting_name' => 'context',
-					'private_setting_value' => $info['widget_context'],
-					'limit' => 0
-				];
-
-				$widgets = elgg_get_entities_from_private_settings($options);
-				/* @var \ElggWidget[] $widgets */
+		if ($info['entity_type'] !== $type) {
```
Comment (jeabakker) [inline engine/lib/widgets.php]: why not use `elgg_extract`

File: engine/lib/widgets.php [hunk 2]
```diff
@@ -303,46 +305,50 @@ function _elgg_create_default_widgets($event, $type, $entity) {
...
+			continue;
+		}
+		
+		if ($info['entity_subtype'] !== ELGG_ENTITIES_ANY_VALUE && $info['entity_subtype'] !== $subtype) {
```
Comment (jeabakker) [inline engine/lib/widgets.php]: same `elgg_extract`

File: engine/lib/widgets.php [hunk 3]
```diff
@@ -303,46 +305,50 @@ function _elgg_create_default_widgets($event, $type, $entity) {
...
-					$new_widget->container_guid = $entity->guid;
-					$new_widget->owner_guid = $entity->guid;
+		// pull in by widget context with widget owners as the site
+		// not using elgg_get_widgets() because it sorts by columns and we don't care right now.
+		$options = [
+			'type' => 'object',
+			'subtype' => 'widget',
+			'owner_guid' => elgg_get_site_entity()->guid,
+			'private_setting_name' => 'context',
+			'private_setting_value' => $info['widget_context'],
```
Comment (jeabakker) [inline engine/lib/widgets.php]: same `elgg_extract`

File: engine/lib/widgets.php [hunk 4]
```diff
@@ -303,46 +305,50 @@ function _elgg_create_default_widgets($event, $type, $entity) {
...
+		foreach ($widgets as $widget) {
+			// change the container and owner
+			$new_widget = clone $widget;
+			$new_widget->container_guid = $entity->guid;
+			$new_widget->owner_guid = $entity->guid;
 
-					$new_widget->save();
-				}
+			// pull in settings
+			$settings = get_all_private_settings($widget->guid);
```
Comment (jeabakker) [inline engine/lib/widgets.php]: why isn't this part of `ElggWidget::clone`

**Review by jeabakker — DISMISSED** [review]

File: engine/lib/widgets.php (hunk 4 above)
Comment (jdalsem) [reply engine/lib/widgets.php]: yep should... not going to fix that as part of this PR

**Pull request merged**